clientcontroller/opstackl2/consumer.go

```diff
@@ -431,6 +433,48 @@ func ConvertProof(cmtProof cmtcrypto.Proof) Proof {
 	}
 }
 
+// GetBlockNumberByTimestamp returns the L2 block number for the given BTC staking activation timestamp.
+// It uses a binary search to find the block number.
+func (cc *OPStackL2ConsumerController) GetBlockNumberByTimestamp(ctx context.Context, targetTimestamp uint64) (uint64, error) {
+	// Check if the target timestamp is after the latest block
+	latestBlock, err := cc.opl2Client.HeaderByNumber(ctx, nil)
+	if err != nil {
+		return 0, err
+	}
+	if targetTimestamp > latestBlock.Time {
+		return 0, nil
```

Comment:
this logic seems incorrect. if the target timestamp is in the future, we will be returning 0 as the activated height. also, I feel we should throw an error instead of nil b/c error I think this is a case where we don't expect it to happen. or am I missing anything?

Reply:
ok, we can return max of uint64 as the activated height and an error.